src/tests/ftest/util/agent_utils.py

```diff
@@ -23,20 +25,15 @@ def include_local_host(hosts):
     """Ensure the local host is included in the specified host list.
 
     Args:
-        hosts (list): list of hosts
+        hosts (NodeSet): list of hosts
 
     Returns:
-        list: list of hosts including the local host
+        NodeSet: list of hosts including the local host
 
     """
-    local_host = socket.gethostname().split('.', 1)[0]
-    if hosts is None:
-        hosts = [local_host]
-    elif local_host not in hosts:
-        # Take a copy of hosts to avoid modifying-in-place
-        hosts = list(hosts)
-        hosts.append(local_host)
-    return hosts
+    with_localhost = NodeSet(socket.gethostname().split('.', 1)[0])
+    with_localhost.add(hosts)
+    return with_localhost
```

Comment:
Just playing with `add` vs `union`...
```suggestion
    return NodeSet(socket.gethostname().split('.', 1)[0]).union(hosts)
```